sql/executor.go

```diff
@@ -39,11 +42,18 @@ var errTransactionInProgress = errors.New("there is already a transaction in pro
 type Executor struct {
 	db     client.DB
 	nodeID uint32
+
+	// System Config and mutex.
+	systemConfig   *config.SystemConfig
```

Comment:
I'll just go ahead and keep the Mutex. It's clear, simple, and doesn't require extra interfaces or use of unsafe.